storage/reads/array_cursor.go

```diff
@@ -63,6 +76,43 @@ func newCountArrayCursor(cur cursors.Cursor) cursors.Cursor {
 	}
 }
 
+func newWindowCountArrayCursor(cur cursors.Cursor, req *datatypes.ReadWindowAggregateRequest) cursors.Cursor {
+	switch cur := cur.(type) {
+	case cursors.FloatArrayCursor:
+		return &integerFloatWindowCountArrayCursor{
+			FloatArrayCursor: cur,
+			every:            req.WindowEvery,
+			tr:               req.Range,
+		}
+	case cursors.IntegerArrayCursor:
+		return &integerIntegerWindowCountArrayCursor{
+			IntegerArrayCursor: cur,
+			every:              req.WindowEvery,
+			tr:                 req.Range,
+		}
+	case cursors.UnsignedArrayCursor:
+		return &integerUnsignedWindowCountArrayCursor{
+			UnsignedArrayCursor: cur,
+			every:               req.WindowEvery,
+			tr:                  req.Range,
+		}
+	case cursors.StringArrayCursor:
+		return &integerStringWindowCountArrayCursor{
+			StringArrayCursor: cur,
+			every:             req.WindowEvery,
+			tr:                req.Range,
+		}
+	case cursors.BooleanArrayCursor:
+		return &integerBooleanWindowCountArrayCursor{
+			BooleanArrayCursor: cur,
+			every:              req.WindowEvery,
+			tr:                 req.Range,
+		}
+	default:
+		panic(fmt.Sprintf("unreachable: %T", cur))
```

Comment:
Is this default really unreachable? Is a default case required for a switch?

Reply:
It's more like an assertion that our _assumption_ that it's unreachable holds. If we get that far, we certainly have a bug, and now we have a stacktrace to show us how we got there.